velox/common/compression/Compression.h

```diff
@@ -17,11 +17,37 @@
 #pragma once
 
 #include <fmt/format.h>
+#include <folly/Expected.h>
 #include <folly/compression/Compression.h>
 #include <string>
 
+#include "velox/common/base/Status.h"
+
 namespace facebook::velox::common {
 
+/// Return with given status wrapped in folly::Unexpected if condition is met.
+#define VELOX_RETURN_UNEXPECTED_IF(condition, status) \
+  do {                                                \
+    if (FOLLY_UNLIKELY(condition)) {                  \
+      return (::folly::makeUnexpected(status));       \
+    }                                                 \
+  } while (false)
+
+/// Propagate any non-successful Status wrapped in folly::Unexpected to the
+/// caller.
+#define VELOX_RETURN_UNEXPECTED_NOT_OK(status)                \
+  do {                                                        \
+    ::facebook::velox::Status __s =                           \
+        ::facebook::velox::internal::genericToStatus(status); \
+    VELOX_RETURN_IF(!__s.ok(), ::folly::makeUnexpected(__s)); \
+  } while (false)
+
+#define VELOX_RETURN_UNEXPECTED(expected)                    \
```

Comment:
don't these belong to velox/common/base/Status.h? Would also be nice to add this as a separate PR, with its own tests